src/main/scala/strawman/collection/WithFilter.scala

```diff
@@ -0,0 +1,76 @@
+package strawman.collection
+
+import scala.{Boolean, Unit}
+
+/** A template trait that contains just the `map`, `flatMap`, `foreach` and `withFilter` methods
+  *  of trait `Iterable`.
+  *
+  * @tparam A Element type (e.g. `Int`)
+  * @tparam C Collection type (e.g. `List[Int]`)
+  */
+class WithFilter[+A, +C](coll: C, iterable: Iterable[A], p: A => Boolean) {
```

Comment:
Here we need both a `coll: C` and an `iterable: Iterable[A]` parameters because `BuildFrom` needs the former to create a builder and we need the latter to iterate through the elements of the collection (we can not get an evidence that `C` is an `Iterable[A]` because that’s not always the case, as explained previously, e.g. think of `StringOps`)